### DISCUSSION TITLE
Upsert method with WHERE clause

### DISCUSSION THREAD

**Discussion by zibi94** [aarondl/sqlboiler#1361] [General]
Let's consider this example:

Our query:
```
INSERT INTO products ("key", "name") 
    VALUES ('1234', 'Product name') 
ON CONFLICT ("key") 
DO UPDATE 
    SET name = EXCLUDED.name 
    WHERE products.name IS NULL 
        AND EXCLUDED.name IS NOT NULL 
RETURNING "key", "name";
```

Execute our query (first execution).
Query return result:
```
key  |     name     
------+--------------
 1234 | Product name
(1 row)

INSERT 0 1
```

Now execute our query again (second execution and each next). 
Query result:
```
key | name 
-----+------
(0 rows)

INSERT 0 0
```

So, if the query executed correctly but did not take action on any row by excluding `WHERE clause`, shouldn't it be handled in some way?

If so, in what way? Return error ? Handle `RowsAffected` like in `Update` method?

**Comment by stephenafamo** [comment]
Not sure 🤷🏾 what to do in cases like this.

I'm also not sure if this is related to SQLBoiler 😅
